tokio-current-thread/src/lib.rs

```diff
@@ -822,6 +909,60 @@ unsafe fn hide_lt<'a>(p: *mut (SpawnLocal + 'a)) -> *mut (SpawnLocal + 'static)
     mem::transmute(p)
 }
 
+
+// ===== impl SpawnHandle =====
+
+// NOTE: It's unfortunate that this code (and the code for `SpawnedWithHandle`)
+// has to be duplicated from `tokio-executor`. However, there was no way for
+// `tokio-current-thread` to construct new `SpawnHandle`s or
+// `SpawnedWithHandle`s without making their constructors public to all users of
+// the `tokio-executor` crate as well.
+impl<T, E> Future for SpawnHandle<T, E> {
```

Comment:
the current-thread crate would not need to add any code after #625. Perhaps we should get that one done first.

Reply:
Alright, I'm happy to put this on hold until then.